cluster/examples/kubernetes/ceph/csi/template/rbd/csi-rbdplugin.yaml

```diff
@@ -23,6 +23,11 @@ spec:
       dnsPolicy: ClusterFirstWithHostNet
       containers:
         - name: driver-registrar
+          # This is necessary only for systems with SELinux, where
+          # non-privileged sidecar containers cannot access unix domain socket
+          # created by privileged CSI driver container.
+          securityContext:
```

Comment:
My understanding of the issue was that only the csi provisioner pods are affected. Were existing volumes failing to mount after the upgrade? If they were mounting successfully, there is no need to change the privs on the plugin pods.

Reply:
Yes some existing volumes mount are failing as the upgrade procedure includes draining for nodes

as the socket file be created by the nodeplugin container in daemonset due to Selinux issue after upgrade the  socket won't be accessible by the driver registrar, its good to run all container as privileged or run all container as unprivileged as daemonset is using bidirectional mount we need  the plugin container to be privileged so we need to make other containers as privileged

Reply:
> Yes some existing volumes mount are failing as the upgrade procedure includes draining for nodes

Why wouldn't all existing volumes be affected? Seems like all should hit this issue or else none.

Reply:
So there is a socket in both the provisioner and the plugin that is affected by this issue? Here is my understanding of the possible fixes. Is this correct? Or please clarify.
1. Run both the container and sidecars as privileged and keep the hostpath
    - This fix should work for both the csi provisioner and plugins 
2. Run both the container and sidecars unprivileged, then change the hostpath to an emptydir
   - This will only work for the provisioner. This won't work for the csi plugins since the hostpath is required

Reply:
> > Yes some existing volumes mount are failing as the upgrade procedure includes draining for nodes
> 
> Why wouldn't all existing volumes be affected? Seems like all should hit this issue or else none.
that's what I heady from QE only two pods are in containerCreating state after upgrade

Reply:
> So there is a socket in both the provisioner and the plugin that is affected by this issue? Here is my understanding of the possible fixes. Is this correct? Or please clarify.
> 
> 1. Run both the container and sidecars as privileged and keep the hostpath
>    
>    * This fix should work for both the csi provisioner and plugins

Yes, as all containers are running as privileged think should work, but  this is not a good fix

> 2. Run both the container and sidecars unprivileged, then change the hostpath to an emptydir
>    
>    * This will only work for the provisioner. This won't work for the csi plugins since the hostpath is required

Yes that's  what my assumption here, in daemonset we have 3 containers as the socket needs to be on the hostpath for communication with kubelet,  we need to run all containers as privileged in daemonset pods

@gnufied can you please confirm this